### PR TITLE
[TE5] Add Python controller command to open enhanced commissioning window for multi-admin

### LINKED ISSUE #9098: Python controller command to open enhanced commissioning window for multi-admin feature.
Provide Python controller command to open enhanced commissioning window for multi-admin feature. Cluster command requires user to provide PAKEVerifier, Salt, Iteration Count and few other parameters. But PAKEVerifier is generated using Salt and Iteration count (and random setup pin code). For cert testing, we need a python command that internally computes PAKEVerifier.

### PR DESCRIPTION
**Pull request by yufengwangca**
#### Problem
What is being fixed?  Examples:
* FR from csg, need to add Python controller command to open enhanced commissioning window for multi-admin feature
* Fixes #9098
#### Change overview
* Add Python controller command to open enhanced commissioning window for multi-admin.

#### Testing
How was this tested? (at least one bullet point required)
* chip-device-ctrl > connect -ip 192.168.86.157 20202021 12344321
* chip-device-ctrl > open-commissioning-window 12344321 -t 3 -o 1 -d 3840 -i 1000
```
[1630448560.815711][1298769:1298777] CHIP:CTL: OperationalDiscoveryComplete for device ID 12344321
[1630448560.817931][1298769:1298777] CHIP:IN: Encrypted message 0x23bfbe0 to 0x0000000000BC5C01 of type 8 and protocolId 5 on exchange 39027.
[1630448560.817946][1298769:1298777] CHIP:IN: Sending msg 0x23bfbe0 to 0x0000000000BC5C01 at utc time: 607023956 msec
[1630448560.817953][1298769:1298777] CHIP:IN: Sending secure msg on generic transport
[1630448560.817971][1298769:1298777] CHIP:IN: Secure msg send status ../../src/inet/IPEndPointBasis.cpp:915: Success
[1630448560.818020][1298769:1298777] CHIP:CTL: Manual pairing code: [36201729767]
[1630448560.818071][1298769:1298777] CHIP:CTL: SetupQRCode: [MT:000006DB00MX1636110]

```

### CODE REVIEW

File: src/controller/CHIPDeviceController.cpp [hunk 1]
```diff
@@ -1041,6 +1043,57 @@ CHIP_ERROR DeviceCommissioner::OperationalDiscoveryComplete(NodeId remoteDeviceI
...
+        pairingWindowOption = Device::PairingWindowOption::kTokenWithProvidedPIN;
+        break;
+    default:
+        ChipLogError(Controller, "Invalid Pairing Window option");
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+
+    err = device->OpenCommissioningWindow(timeout, iteration, pairingWindowOption, salt, payload);
+
+    if (err != CHIP_NO_ERROR && pairingWindowOption != Device::PairingWindowOption::kOriginalSetupCode)
```
[the remark this answers is not in the document]
Comment (yufengwangca) [reply src/controller/CHIPDeviceController.cpp]: Good catch!

File: src/controller/CHIPDeviceController.cpp [hunk 2]
```diff
@@ -1039,6 +1041,57 @@ CHIP_ERROR DeviceCommissioner::OperationalDiscoveryComplete(NodeId remoteDeviceI
...
+        break;
+    case 2:
+        pairingWindowOption = Device::PairingWindowOption::kTokenWithProvidedPIN;
+        break;
+    default:
+        ChipLogError(Controller, "Invalid Pairing Window option");
+        return CHIP_ERROR_INVALID_ARGUMENT;
+    }
+
+    err = device->OpenCommissioningWindow(timeout, iteration, pairingWindowOption, salt, payload);
```
Comment (pan-apple) [inline src/controller/CHIPDeviceController.cpp:1079]: nit: you can use `ReturnErrorOnFailure`, and avoid the error check in next line.

File: src/controller/CHIPDeviceController.cpp [hunk 3]
```diff
@@ -1039,6 +1041,57 @@ CHIP_ERROR DeviceCommissioner::OperationalDiscoveryComplete(NodeId remoteDeviceI
...
+
+    if (err == CHIP_NO_ERROR && pairingWindowOption != Device::PairingWindowOption::kOriginalSetupCode)
+    {
+        err = ManualSetupPayloadGenerator(payload).payloadDecimalStringRepresentation(manualPairingCode);
```
Comment (pan-apple) [inline src/controller/CHIPDeviceController.cpp:1083]: nit: Use `ReturnErrorOnFailure`?

File: src/controller/CHIPDeviceController.cpp [hunk 4]
```diff
@@ -1039,6 +1041,57 @@ CHIP_ERROR DeviceCommissioner::OperationalDiscoveryComplete(NodeId remoteDeviceI
...
+        VerifyOrReturnError(err == CHIP_NO_ERROR, err);
+        ChipLogProgress(Controller, "Manual pairing code: [%s]", manualPairingCode.c_str());
+
+        err = QRCodeSetupPayloadGenerator(payload).payloadBase38Representation(QRCode);
```
Comment (pan-apple) [inline src/controller/CHIPDeviceController.cpp:1087]: nit: Use `ReturnErrorOnFailure`?

**Review by pan-apple — APPROVED** [review]
Minor nits for code cleanup and consistency.
Code looks ok otherwise.

**Review by jmartinez-silabs — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

**Review by hawk248 — APPROVED** [review]
👍

**Review by chrisdecenzo — APPROVED** [review]

**Comment by andy31415** [comment]
Cherrypick conflict in src/controller/python/chip/ChipDeviceCtrl.py

**Comment by andy31415** [comment]
Manually resolved the conflict - kept the commision window open command, dropped the 'GetCompressedFabricId' one.

**Pull request merged**